sycl/unittests/kernel-and-program/KernelInfo.cpp

```diff
@@ -1,139 +0,0 @@
-  return PI_SUCCESS;
-}
-
-static pi_result
-redefinedProgramBuild(pi_program program, pi_uint32 num_devices,
-                      const pi_device *device_list, const char *options,
-                      void (*pfn_notify)(pi_program program, void *user_data),
-                      void *user_data) {
-  return PI_SUCCESS;
-}
-
-static pi_result redefinedKernelCreate(pi_program program,
-                                       const char *kernel_name,
-                                       pi_kernel *ret_kernel) {
-  return PI_SUCCESS;
-}
-
-static pi_result redefinedKernelRetain(pi_kernel kernel) { return PI_SUCCESS; }
-
-static pi_result redefinedKernelRelease(pi_kernel kernel) { return PI_SUCCESS; }
-
-static pi_result redefinedKernelGetInfo(pi_kernel kernel,
-                                        pi_kernel_info param_name,
-                                        size_t param_value_size,
-                                        void *param_value,
-                                        size_t *param_value_size_ret) {
-  EXPECT_EQ(param_name, PI_KERNEL_INFO_CONTEXT)
-      << "Unexpected kernel info requested";
-  auto *Result = reinterpret_cast<RT::PiContext *>(param_value);
-  RT::PiContext PiCtx =
-      detail::getSyclObjImpl(TestContext->Ctx)->getHandleRef();
-  *Result = PiCtx;
-  return PI_SUCCESS;
-}
-
-static pi_result redefinedKernelSetExecInfo(pi_kernel kernel,
-                                            pi_kernel_exec_info param_name,
-                                            size_t param_value_size,
-                                            const void *param_value) {
-  return PI_SUCCESS;
-}
-
-class KernelInfoTest : public ::testing::Test {
-public:
-  KernelInfoTest() : Plt{default_selector()} {}
-
-protected:
-  void SetUp() override {
-    if (Plt.is_host()) {
-      std::clog << "This test is only supported on non-host platforms.\n";
-      std::clog << "Current platform is "
-                << Plt.get_info<info::platform::name>();
-      return;
-    }
-
-    Mock = std::make_unique<unittest::PiMock>(Plt);
-
-    Mock->redefine<detail::PiApiKind::piKernelGetGroupInfo>(
-        redefinedKernelGetGroupInfo);
-    Mock->redefine<detail::PiApiKind::piclProgramCreateWithSource>(
-        redefinedProgramCreateWithSource);
-    Mock->redefine<detail::PiApiKind::piProgramBuild>(redefinedProgramBuild);
-    Mock->redefine<detail::PiApiKind::piKernelCreate>(redefinedKernelCreate);
-    Mock->redefine<detail::PiApiKind::piKernelRetain>(redefinedKernelRetain);
-    Mock->redefine<detail::PiApiKind::piKernelRelease>(redefinedKernelRelease);
-    Mock->redefine<detail::PiApiKind::piKernelGetInfo>(redefinedKernelGetInfo);
-    Mock->redefine<detail::PiApiKind::piKernelSetExecInfo>(
-        redefinedKernelSetExecInfo);
-  }
-
-protected:
-  platform Plt;
-  std::unique_ptr<unittest::PiMock> Mock;
-};
-
-TEST_F(KernelInfoTest, GetPrivateMemUsage) {
```

Comment:
Could this potentially be adapted to use `kernel_bundle`? If not now, then in the near future?